client/examples/retry_client.rs

```diff
@@ -22,25 +22,24 @@ pub struct RetryClient {
 const INTERVAL: u64 = 1000;
 const RETRY_ATTEMPTS: u8 = 10;
 
+#[async_trait]
```

Comment:
Is it possible to drop this dependency by just writing the functions more verbose using `Future`?

Reply:
You mean `Pin<Box<dyn Future<...> + Send +'_>>` (https://smallcultfollowing.com/babysteps/blog/2019/10/26/async-fn-in-traits-are-hard/)

Reply:
I think it may be a little bit more involved, see: https://github.com/dtolnay/async-trait#explanation